hazelcast-client/src/test/java/classloading/ThreadLeakClientTest.java

```diff
@@ -42,13 +55,69 @@ public void testThreadLeak() {
     @Test
     public void testThreadLeak_withoutCluster() {
         ClientConfig clientConfig = new ClientConfig();
-        clientConfig.getConnectionStrategyConfig()
-                .setAsyncStart(true);
-        clientConfig.getNetworkConfig()
-                .setConnectionAttemptLimit(Integer.MAX_VALUE);
+        clientConfig.getConnectionStrategyConfig().setAsyncStart(true);
+        clientConfig.getNetworkConfig().setConnectionAttemptLimit(Integer.MAX_VALUE);
 
         HazelcastInstance client = HazelcastClient.newHazelcastClient(clientConfig);
 
         client.shutdown();
     }
+
+    @Test (expected = IllegalStateException.class)
+    public void testThreadLeakWhenClientCanNotStart() {
+        HazelcastClient.newHazelcastClient();
+        Assert.fail();
+    }
+
+    @Test (expected = IllegalStateException.class)
+    public void testThreadLeakWhenClientCanNotStartDueToAuthenticationError() {
+        try {
+            Hazelcast.newHazelcastInstance();
+            ClientConfig config = new ClientConfig();
+            config.getGroupConfig().setName("invalid cluster");
+            HazelcastClient.newHazelcastClient(config);
+            Assert.fail();
+        } finally {
+            Hazelcast.shutdownAll();
+        }
+    }
+
+    @Test (expected = IllegalStateException.class)
+    public void testThreadLeakWhenClientCanNotConstructDueToNoMemberDiscoveryStrategyConfig() {
+        ClientConfig config = new ClientConfig();
+        config.getNetworkConfig().getDiscoveryConfig().addDiscoveryStrategyConfig(
+                new DiscoveryStrategyConfig(new ClientDiscoverySpiTest.NoMemberDiscoveryStrategyFactory(),
+                        Collections.<String, Comparable>emptyMap()));
+        HazelcastClient.newHazelcastClient(config);
+        Assert.fail();
```

Comment:
This statement is superfluous. Because you said 'expected = ', the Junit framework will fail this test if it doesn't exit with the expected exception.

Reply:
Ok, removed those lines.